### PR TITLE
[FLINK-9249][build] Add convenience profile for skipping non-essential plugins

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

This PR adds a convenience profile for skipping non-essential plugins when building Flink.
Disabled plugins:
* rat
* checkstyle
* scalastyle
* enforcer
* japicmp
* javadoc

The PR also modifies the travis script to make use of the profile.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -636,6 +636,63 @@ under the License.
...
+							<configuration>
+								<skip>true</skip>
+							</configuration>
+						</plugin>
+						<plugin>
+							<groupId>org.apache.maven.plugins</groupId>
+							<artifactId>maven-checkstyle-plugin</artifactId>
+							<configuration>
+								<skip>true</skip>
+							</configuration>
```
Comment (zhangminglei) [inline pom.xml]: If we skip this plugin, Will it affect the format or style of the code of people's PR will merge to flink?

Comment (W4anD0eR96) [reply pom.xml]: +1 not skip plugins on tarvis. LGTM make it fast local building.

Comment (zentol) [reply pom.xml]: we still verify everything at least once on travis.

Comment (zhangminglei) [reply pom.xml]: That's good. I think this works because of ```.travis.yml``` enable it at least. Correct ?

Comment (zentol) [reply pom.xml]: got nothing to do `.travis.yml`. The `travis_mvn_watchdog.sh` script uses differet settings for each build, one of which builds all of Flink with all plugins enabled, (so called "misc.")

Comment (zhangminglei) [reply pom.xml]: That sounds nice.

**Comment by bowenli86** [comment]
+1

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**